Kernel/Term.hpp

```diff
@@ -88,10 +88,8 @@ enum ArgumentOrderVals {
   AO_UNKNOWN=0,
   AO_GREATER=1,
   AO_LESS=2,
-  AO_GREATER_EQ=3,
-  AO_LESS_EQ=4,
-  AO_EQUAL=5,
-  AO_INCOMPARABLE=6,
+  AO_EQUAL=3,
+  AO_INCOMPARABLE=4,
```

Comment:
It's slightly annoying that this still needs 5 values. Otherwise we could save *one whole bit* (gasp) in terms.

Are things only `AO_EQUAL` syntactically? If so I could imagine `AO_EQUAL_OR_UNKNOWN` and a fast equality check to differentiate `AO_UNKNOWN` and `AO_EQUAL`. But this is premature optimisation.

Reply:
I think so. But let's keep that for another PR - I'll put it on my list to investigate 'eventually' unless you get inspired. :-)